src/windows/win-crowdstrike-fix-bootloop-v2.ps1

```diff
@@ -75,15 +75,48 @@ function LoadUnloadRegistryHives
                     $regFile = $regKeyToFile[$regKey]
                     if (Test-Path -Path $regFile) {
                         Log-Info "Found registry config file at $regFile."
-                        $configExist = $true
+                        $configExistInDrive = $true
 
                         Log-Info "Loading registry hive $regKey from $regFile..."
+                        $succeeded = $false
                         $result = reg load $regKey $regFile 2>&1
                         if ($LASTEXITCODE -ne 0) {
                             Log-Error "Load registry hive $regKey from $regFile failed with exit code $LASTEXITCODE. Error: $result"
-                        } else {
+                            if ($result -Match "corrupt") {
+                                Log-Info "Making a backup of the original registry config file..."
+                                $timestamp = Get-Date -Format "yyyyMMddTHHmmssffffZ"
+                                $backupFileName = "$regFile.backup-$timestamp"
+                                try {
+                                    Copy-Item $regFile -Destination $backupFileName -ErrorAction Stop
+                                    Log-Info "Original registry config file $regFile is backed up at $backupFileName"
```

Comment:
might be good to move the backup code within the FixRegistryCorruption function for readability as this function has become too long (and backup can be an implicit step of fix.

Reply:
Sounds good. Moved.